modules/policy/engine/src/main/java/org/eclipse/sw360/antenna/policy/engine/PolicyEngine.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) Bosch Software Innovations GmbH 2019.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.antenna.policy.engine;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+/**
+ * Policy engine that executes all rules on the given data by calling the different {@link RuleExecutor} instances.
+ */
+public class PolicyEngine {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PolicyEngine.class);
+
+    private final Collection<RuleExecutor> executors;
```

Comment:
If we just use `Collection` here, the iteration order is not necessarily the insertion order. I wonder whether we *do* want to give the caller control over the iteration / execution order, e.g. by using a `List` here. I'm aware you're anyway using `parallelStream()` below, so order of execution is non-deterministic currently anyway, so maybe `evaluate()` could get a boolean argument to toggle parallelism or not.

Basically, this boils down to a design decision, and I'd be fine with never guaranteeing execution order, but I feel this should be clearly documented. In any case I'm missing documentation in this class.

Reply:
There is no sequence in the execution of executors. So the collection is the right data structures. All rules are executed in parallel and are fed with all relevant combinations of artifacts. I do not see a logical sequence in Rule execution as of now as a concept.